### ISSUE TITLE
grunt test fails with node 8.1, phantoms 2.1

### ISSUE THREAD

**Issue by thatcher** [openseadragon/openseadragon#1331]
Tests fail to run.

## Symptom:
```
$ grunt test --stack
...
Running "qunit:normal" (qunit) task
Testing http://localhost:8000/test/test.html 
PhantomJS threw an error:ERROR
>> 0 [ '' ]
Warning: PhantomJS exited unexpectedly with exit code null. Use --force to continue.
```

When run with `--force`:

```
$ grunt test --stack --force
...
PhantomJS threw an error:ERROR
>> 0 [ '' ]
Warning: PhantomJS exited unexpectedly with exit code null. Used --force, continuing.
Warning: 0/0 assertions ran (0ms) Used --force, continuing.
(node:37480) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
```

## Environment (Mac OSx 10.12.6):
```
$ node --version
v8.1.0
$ phantomjs --version
2.1.1
$ grunt --version
grunt-cli v1.2.0
grunt v0.4.5
```

Is there a specific phantomjs or node version ceiling?  Are there some libraries we could update?

Thanks,
Thatcher

**Comment by iangilman** [comment]
Interesting! I'm getting the same result. Travis is still running it successfully (see https://travis-ci.org/openseadragon/openseadragon). Anyway, we should definitely fix it! I'm guessing maybe just upgrading `qunitjs` and/or `grunt-qunit-istanbul`. Would you be up for giving that a try? 

FWIW, here is my environment (I'm also on Mac 10.12.6): 

```
$ node --version
v6.9.5
$ phantomjs --version
-bash: phantomjs: command not found
$ grunt --version
grunt-cli v0.1.6
grunt v0.4.5
```

Looks like I don't have phantom installed globally; it must just be using the local install from the package.json... I assume it's installed by grunt-qunit-instanbul or something.

**Comment by thatcher** [comment]
Yes I will work on this one today, thanks Ian.

**Comment by iangilman** [comment]
Wonderful, thank you!

**Comment by nein09** [comment]
I started looking at this because I want to be able to run some tests. The good news is that it's mostly QUnit API/syntax changes that need to take place in order to get us going. The bad news is that`grunt-qunit-istanbul` no longer works with the latest QUnit, so we'll have to find some other way of doing code coverage.

**Comment by nein09** [comment]
"we renamed everything" : http://qunitjs.com/upgrade-guide-2.x/

**Comment by iangilman** [comment]
@nein09 Thank you for digging into this! Any thoughts on what we could use for code coverage? It's not a big deal if we're without for a while, but it would be good to get it working again at some point. Fixing the tests themselves is definitely higher priority.

**Comment by nein09** [comment]
@iangilman I'll have to experiment, but I saw a little demo of qunit outputting something that I think coveralls could use. If it's not a ton of work, I'll put that in this fix.

**Comment by iangilman** [comment]
Sounds good! Thank you for looking into it. I know very little about this stuff...

**Comment by iangilman** [comment]
Fixed by #1374.
